### ISSUE TITLE
[Feature Request] _return per-atom style variables from python function

### ISSUE THREAD

**Issue by thangckt** [lammps/lammps#1566]
**Summary**

Is there any way to return per-atom variable as calling a python function?

**Detailed Description**
Dear Lammps developers,

It will be extremely useful if there is a capability to return per-atom variables after the call of python function. I hope that this can be implemented in the future LAMMPS packages

Thank you so much.

**Comment by akohlmey** [comment]
> **Summary**
> 
> Is there any way to return per-atom variable as calling a python function?

no. those are similar in concept to equal style variables.

> **Detailed Description**
> Dear Lammps developers,
> 
> It will be extremely useful if there is a capability to return per-atom variables after the call of python function. I hope that this can be implemented in the future LAMMPS packages

if you are already programming in python, then there is little incentive to convert the output to an atom style like format. rather you can use the python interface to LAMMPS to process per atom data directly (and in parallel) and then have that python code executed through fix python/invoke or similar.

i should also point out, that it is usually not a good idea to use python code for anything too complex and sophisticated (outside of prototyping), as it incurs severe performance penalties and when implementing the same in C++, access to data is significantly simpler to realize.

most examples that i have seen, where people want to do what you describe, they serialize the processing of data, which is a further problem and not at all desirable in a software like LAMMPS, that is designed to parallelize well.

in summary:
- for simple cases, you can process per atom data through the library interface (using fix property/atom for custom storage, if needed) and use other methods that a variable to trigger execution
- for more complex cases, you should consider writing C++ code, since the overhead of the python code will become significant and for per-atom data, things have to be processed properly in parallel
- if these two options are not satisfactory, then it is likely a design choice problem, and things might work better, by having python drive LAMMPS through the python wrapper instead of LAMMPS driving python through python style variables. 

> 
> Thank you so much.

**Comment by akohlmey** [comment]
at any rate, if you want somebody to look into this seriously, you have to motivate your request with a convincing use case.

**Comment by akohlmey** [comment]
no reaction demonstrating a meaningful use case for 3 weeks. will close the issue within the next week if there is still no reaction.

**Comment by akohlmey** [comment]
closed due to inactivity
